### ISSUE TITLE
Use tolerance also with containsEntry and similar methods

### ISSUE THREAD

**Issue by apozo-copyright** [lukas-krecan/JsonUnit#512]
BigDecimals comparison is a real pain. `ContainsEntry` method is very useful to concatenate checks, but I found that it seems to use equals instead of compareTo, and the tolerance config is neither applied. This fails and it should work:
`assertThatJson("{\"a\":1, \"b\":2.0}").withTolerance(0).isObject().containsEntry("a", BigDecimal.ONE).containsEntry("b", BigDecimal.valueOf(2));`

**Comment by lukas-krecan** [comment]
Hi, thanks for feedback. This is tricky, when I implemented the integration with AssertJ I did not see all the possible edge cases and ways how you would want to use the library. In this case it's necessay to help JsonUnit a bit and say that the value is supposed to be compred as JSON not as a number this way. If you do not do that, the standard AssertJ comparison is used and the BigDecimals do not match. I will check the doc to see if it can be improved in this regard.

```
        assertThatJson("{\"a\":1, \"b\":2.0}")
            .withTolerance(0)
            .isObject()
            .containsEntry("a", json(1))
            .containsEntry("b", json(2));
```

**Comment by lukas-krecan** [comment]
I have been thinking about it and I added special handling for numbers in this case. Once I release a new version, this should work

```
 assertThatJson("{\"a\":1, \"b\":2.0}")
            .withTolerance(0)
            .isObject()
            .containsEntry("a", 1)
            .containsEntry("b", 2);
```

**Comment by apozo-copyright** [comment]
Thank you very much!

**Comment by lukas-krecan** [comment]
Released as 2.35.0
